src/lib/io/sol-iio.c

```diff
@@ -1107,10 +1112,12 @@ channel_get_offset(struct sol_iio_channel *channel)
 
 #define GET_OFFSET(_name) \
     do { \
-        r = craft_filename_path(path, sizeof(path), CHANNEL_OFFSET_PATH, \
-            device->device_id, _name); \
-        if (r && sol_util_read_file(path, "%d", &channel->offset) > 0) { \
-            goto end; \
+        if (_name) { \
+            r = craft_filename_path(path, sizeof(path), CHANNEL_OFFSET_PATH, \
+                device->device_id, _name); \
+            if (r && sol_util_read_file(path, "%d", &channel->offset) > 0) { \
+                goto end; \
+            } \
```

Comment:
@ceolin Thanks for your comment. The channel_get_offset was called by sol_iio_add_channel.  The channle->name should be !NULL in all cases.  But adding the checking is always safe. Let me add the NULL checking before GET_OFFSET call